Libraries/LibWeb/HTML/RenderingThread.h

```diff
@@ -20,17 +21,18 @@ class RenderingThread {
     AK_MAKE_NONMOVABLE(RenderingThread);
 
 public:
-    RenderingThread();
+    RenderingThread(TraversableNavigable&);
     ~RenderingThread();
 
     void start();
     void set_skia_player(OwnPtr<Painting::DisplayListPlayerSkia>&& player) { m_skia_player = move(player); }
-    void enqueue_rendering_task(NonnullRefPtr<Painting::DisplayList>, NonnullRefPtr<Gfx::PaintingSurface>, Function<void()>&& callback);
+    void enqueue_rendering_task(NonnullRefPtr<Painting::DisplayList>, NonnullRefPtr<Painting::BackingStore>, Function<void()>&& callback);
 
 private:
     void rendering_thread_loop();
 
     Core::EventLoop& m_main_thread_event_loop;
+    TraversableNavigable& m_traversable_navigable;
```

Comment:
I expect this to need a GC plugin annotation, since you're holding onto a GC-allocated type by ref.

Reply:
indeed, I am going to mark it as IGNORE_GC because it should be safe to assume traversable will live for as long as rendering thread is working.

Reply:
Also, this may be a naive question, but can there only be one traversable navigable / per render thread? Or i guess this is the "top level traversable"?

Reply:
as I recall all traversables are top level. there is 1:1 relationship between rendering threads and traversable navigables. webcontent process could have multiple traversables, which means multiple rendering threads. but there is only one Skia GPU context shared between all traversable within WebContent, probably we will have to change that.